doc/index.rst

```diff
@@ -721,6 +744,29 @@ Functions
 
    .. versionadded:: 4.1.0
 
+
+Helper classes
+~~~~~~~~~~~~~~
+
+.. class:: DocInfo(documentation)
+
+   Container for documentation information as returned by :func:`doc`.
+
+   It is expected this class wouldn't be manually created but instead returned
+   by :func:`doc` inside of type annotations using :data:`Annotated`.
+
+   The ``documentation`` attribute contains the documentation string passed
+   to :func:`doc`.
+
+   .. versionadded:: 4.8.0
+
+      See :pep:`727`.
+
+   .. attribute:: documentation
+
+      The documentation string passed to :func:`doc`.
```

Comment:
I'm on it.

Is the docstring of the class itself okay? Or should I use

```
:data:`Annotated`
```

instead of:

```
`Annotated`
```
?

Reply:
ReST markup is preferred here, so that Sphinx can add a link to the docs for `Annotated`. But it doesn't make sense to include ReST markup in the docstring, since we don't autogenerate docs from the docstring 👍

Reply:
Ah, perfect. Thank you.

Reply:
No problem, thanks for the update!